components/brave_wallet/browser/swap_service.cc

```diff
@@ -113,15 +180,28 @@ GURL Append0xSwapParams(const GURL& swap_url,
   if (!params.sell_token.empty()) {
     url = net::AppendQueryParameter(url, "sellToken", params.sell_token);
   }
-  url =
-      net::AppendQueryParameter(url, "buyTokenPercentageFee", GetFee(chain_id));
+
+  auto brave_swap_fee_params = brave_wallet::mojom::BraveSwapFeeParams::New();
+  brave_swap_fee_params->chain_id = chain_id;
+  brave_swap_fee_params->taker = params.taker_address;
+  brave_swap_fee_params->input_token = params.sell_token;
+  brave_swap_fee_params->output_token = params.buy_token;
+  const auto& brave_fee = GetBraveFeeInternal(std::move(brave_swap_fee_params));
+  double effective_fee_pct;
```

Comment:
Let's keep numbers on stack initialized
`  double effective_fee_pct = 0.0;`
and below also